codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttDecoder.java

```diff
@@ -402,6 +556,25 @@ private static MqttFixedHeader decodeFixedHeader(ByteBuf buffer) {
         return new Result<MqttSubAckPayload>(new MqttSubAckPayload(grantedQos), numberOfBytesConsumed);
     }
 
+    private static Result<MqttUnsubAckPayload> decodeUnsubAckPayload(
+            ChannelHandlerContext ctx,
+            ByteBuf buffer,
+            int bytesRemainingInVariablePart) {
+        final MqttVersion mqttVersion = MqttCodecUtil.getMqttVersion(ctx);
+        if (mqttVersion == MqttVersion.MQTT_5) {
+            final List<Short> reasonCodes = new ArrayList<Short>();
```

Comment:
nit: use `new ArrayList<Short>(bytesRemainingInVariablePart);` as we already know how many elements we will add.